### ISSUE TITLE
Tailwind not parse the whole template component

### ISSUE THREAD

**Issue by lunoob** [resendlabs/react-email#1021]
### Describe the Bug

Before version 0.0.8, the following code is working normally
```tsc
<Tailwind>
  <Template />
</Tailwind>
```

After version 0.0.8, I found that it only analyzes Component with the Children attribute
```tsc
// it work fine
<Tailwind>
   <Component className="text-[100px]">React Email</Component>
</Tailwind>

// it not work for <div>
<Tailwind>
   <Component className="text-[100px]">
        <div className="text-red">Color</div>
        React Email
   </Component>
</Tailwind>
```

### Which package is affected (leave empty if unsure)

@react-email/tailwind

### Link to the code that reproduces this issue

[example in codesandbox](https://codesandbox.io/p/sandbox/vibrant-margulis-crq8hm?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clonm98zy000a3b6i81afg7gi%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clonm98zy00033b6iapcuchic%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clonm98zy00073b6ibopextq6%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clonm98zy00093b6igdyq5oyu%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clonm98zy00033b6iapcuchic%2522%253A%257B%2522id%2522%253A%2522clonm98zy00033b6iapcuchic%2522%252C%2522activeTabId%2522%253A%2522clonmbkc700np3b6ijcp8wzb3%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fmain.tsx%2522%252C%2522id%2522%253A%2522clonma8no00ci3b6iov0q59i8%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522id%2522%253A%2522clonmabcu00g83b6inveu59ij%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fcomponents%252FEmailTemplate%252Findex.tsx%2522%252C%2522id%2522%253A%2522clonmbkc700np3b6ijcp8wzb3%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fcomponents%252FEmailTemplate%252FRoot.tsx%2522%252C%2522id%2522%253A%2522clonmgh5j03by3b6iddxf0kkf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fcomponents%252FEmailTemplate%252FBrand.tsx%2522%252C%2522id%2522%253A%2522clonmnp4805y23b6isn9t9qa4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clonm98zy00093b6igdyq5oyu%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clonm98zy00083b6i1866vpy0%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clonm98zy00093b6igdyq5oyu%2522%252C%2522activeTabId%2522%253A%2522clonm98zy00083b6i1866vpy0%2522%257D%252C%2522clonm98zy00073b6ibopextq6%2522%253A%257B%2522id%2522%253A%2522clonm98zy00073b6ibopextq6%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clonm98zy00043b6iidsmudz1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clonm98zy00053b6ibl66efcf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522build%2522%257D%252C%257B%2522id%2522%253A%2522clonm98zy00063b6i8d3w1k1f%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522preview%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clonm98zy00043b6iidsmudz1%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### To Reproduce

I compared the latest version and the source code of version 0.0.8, and found that the 0.0.8 version was replaced by the entire HTML, while the latest version only analyzes Children after the version 0.0.8.

0.0.8
![image](https://github.com/resendlabs/react-email/assets/49156607/b6b11103-e0ee-4387-ae5b-ad1f8edf8177)

0.0.8+
![image](https://github.com/resendlabs/react-email/assets/49156607/322ab1b6-61ab-4437-8388-ebcfb1a996e6)

### Expected Behavior

it can be parse the whole template component

**Comment by gabrielmfern** [comment]
Can you maybe elaborate more on what is happening to your email template? Does it work under some circumstances and others not, and if so, can you please tell me what circumstances?

**Comment by giovannetti-eric** [comment]
Same problem here, with `@react-email/components": "0.0.7"` my email templates were working well, after upgrade, a lot of Tailwind classes are not converted into style attributes anymore.

For example:

``` jsx
  <Link href={t("common.appStore.href")} className="inline-block">
    <Img
      src={t("common.appStore.img")}
      width="119"
      height="40"
      alt={t("common.appStore.label")}
      className="max-w-full h-auto inline"
    />
  </Link>
```

is generating

``` html
<a href="xxx" class="inline-block" style="color:#067df7;text-decoration:none" target="_blank"><img class="max-w-full h-auto inline" alt="xxx" height="40" src="xxx" style="display:block;outline:none;border:none;text-decoration:none" width="119"></a>
```

After some test, I found that the problem apply when I load custom components in an email template, for example:

``` tsx
import * as React from "react";
import MainLayout from "../components/ui/MainLayout";
import Email1 from "../components/views/Email1";
import { useGetToken } from "../hooks/useGetToken";

const locale = "en";

export const Email1En = () => {
  const { t } = useGetToken(locale);

  return (
    <MainLayout
      title={t("pnp23bf1_22.title")}
      preview={t("pnp23bf1_22.preview")}
    >
      <Email1 locale={locale} />
    </MainLayout>
  );
};

export default Email1En;
```

``` tsx
import { Body, Head, Html, Preview } from "@react-email/components";
import React from "react";
import TailwindProvider from "./TailwindProvider";

interface MainLayoutProps {
  title?: string;
  preview?: string;
  children?: React.ReactNode;
}

export const MainLayout = ({ title, preview, children }: MainLayoutProps) => {
  return (
      <Html>
        <Head>
          <title>{title}</title>
          <link rel="preconnect" href="https://fonts.googleapis.com" />
          <link
            rel="preconnect"
            href="https://fonts.gstatic.com"
            crossOrigin="true"
          />
          <link
            href="https://fonts.googleapis.com/css2?family=Roboto+Slab&display=swap"
            rel="stylesheet"
          />
          <style>
            {`
              @media screen and (max-width: 600px) {
                h1 {
                  font-size: 30px !important;
                }
              }
            `}
          </style>
        </Head>
        {preview && <Preview>{preview}</Preview>}
        <TailwindProvider>
          <Body className="text-gray-900 bg-white my-auto mx-auto font-sans leading-snug">
              {children}
          </Body>
        </TailwindProvider>
      </Html>
  );
};

export default MainLayout;
```

Most Tailwind classes into `<Email1 />` are not generated

**Comment by lunoob** [comment]
> Can you maybe elaborate more on what is happening to your email template? Does it work under some circumstances and others not, and if so, can you please tell me what circumstances?

yes, I make a [example](https://codesandbox.io/p/sandbox/vibrant-margulis-crq8hm?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clonm98zy000a3b6i81afg7gi%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clonm98zy00033b6iapcuchic%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clonm98zy00073b6ibopextq6%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clonm98zy00093b6igdyq5oyu%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clonm98zy00033b6iapcuchic%2522%253A%257B%2522id%2522%253A%2522clonm98zy00033b6iapcuchic%2522%252C%2522activeTabId%2522%253A%2522clonmbkc700np3b6ijcp8wzb3%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fmain.tsx%2522%252C%2522id%2522%253A%2522clonma8no00ci3b6iov0q59i8%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522id%2522%253A%2522clonmabcu00g83b6inveu59ij%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fcomponents%252FEmailTemplate%252Findex.tsx%2522%252C%2522id%2522%253A%2522clonmbkc700np3b6ijcp8wzb3%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fcomponents%252FEmailTemplate%252FRoot.tsx%2522%252C%2522id%2522%253A%2522clonmgh5j03by3b6iddxf0kkf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fcomponents%252FEmailTemplate%252FBrand.tsx%2522%252C%2522id%2522%253A%2522clonmnp4805y23b6isn9t9qa4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clonm98zy00093b6igdyq5oyu%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clonm98zy00083b6i1866vpy0%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clonm98zy00093b6igdyq5oyu%2522%252C%2522activeTabId%2522%253A%2522clonm98zy00083b6i1866vpy0%2522%257D%252C%2522clonm98zy00073b6ibopextq6%2522%253A%257B%2522id%2522%253A%2522clonm98zy00073b6ibopextq6%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clonm98zy00043b6iidsmudz1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clonm98zy00053b6ibl66efcf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522build%2522%257D%252C%257B%2522id%2522%253A%2522clonm98zy00063b6i8d3w1k1f%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522preview%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clonm98zy00043b6iidsmudz1%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by gabrielmfern** [comment]
Going to take a look into fixing this to unblock you guys, after fixing this a PR is coming that'll improve a lot of things about the Tailwind component.

**Comment by Maclay74** [comment]
I think it might be related with recent react update. Essentially, they "fixed bug" when rendering functions take context into account.

**Comment by gabrielmfern** [comment]
> Same problem here, with `@react-email/components": "0.0.7"` my email templates were working well, after upgrade, a lot of Tailwind classes are not converted into style attributes anymore.
> 
> For example:
> 
> ```js
>   <Link href={t("common.appStore.href")} className="inline-block">
>     <Img
>       src={t("common.appStore.img")}
>       width="119"
>       height="40"
>       alt={t("common.appStore.label")}
>       className="max-w-full h-auto inline"
>     />
>   </Link>
> ```
> 
> is generating
> 
> ```
> <a href="xxx" class="inline-block" style="color:#067df7;text-decoration:none" target="_blank"><img class="max-w-full h-auto inline" alt="xxx" height="40" src="xxx" style="display:block;outline:none;border:none;text-decoration:none" width="119"></a>
> ```
> 
> After some test, I found that the problem apply when I load custom components in an email template, for example:
> 
> ```tsx
> import * as React from "react";
> import MainLayout from "../components/ui/MainLayout";
> import Email1 from "../components/views/Email1";
> import { useGetToken } from "../hooks/useGetToken";
> 
> const locale = "en";
> 
> export const Email1En = () => {
>   const { t } = useGetToken(locale);
> 
>   return (
>     <MainLayout
>       title={t("pnp23bf1_22.title")}
>       preview={t("pnp23bf1_22.preview")}
>     >
>       <Email1 locale={locale} />
>     </MainLayout>
>   );
> };
> 
> export default Email1En;
> ```
> 
> ```tsx
> import { Body, Head, Html, Preview } from "@react-email/components";
> import React from "react";
> import TailwindProvider from "./TailwindProvider";
> 
> interface MainLayoutProps {
>   title?: string;
>   preview?: string;
>   children?: React.ReactNode;
> }
> 
> export const MainLayout = ({ title, preview, children }: MainLayoutProps) => {
>   return (
>       <Html>
>         <Head>
>           <title>{title}</title>
>           <link rel="preconnect" href="https://fonts.googleapis.com" />
>           <link
>             rel="preconnect"
>             href="https://fonts.gstatic.com"
>             crossOrigin="true"
>           />
>           <link
>             href="https://fonts.googleapis.com/css2?family=Roboto+Slab&display=swap"
>             rel="stylesheet"
>           />
>           <style>
>             {`
>               @media screen and (max-width: 600px) {
>                 h1 {
>                   font-size: 30px !important;
>                 }
>               }
>             `}
>           </style>
>         </Head>
>         {preview && <Preview>{preview}</Preview>}
>         <TailwindProvider>
>           <Body className="text-gray-900 bg-white my-auto mx-auto font-sans leading-snug">
>               {children}
>           </Body>
>         </TailwindProvider>
>       </Html>
>   );
> };
> 
> export default MainLayout;
> ```
> 
> Most Tailwind classes into `<Email1 />` are not generated

Working on a fix for this on a new PR that also improves performance.

The problem here is that only the children passed through props to components are handled,
while the resulting children of the rendered component are not.

This is indeed a bug and we are on it.

**Comment by gabrielmfern** [comment]
PR that solves this is basically done, this problem will go away soon guys. 

If it takes too long for us to merge and release a new version out, I made a quick patch that fixes the issue as a workaround

````patch
diff --git a/dist/index.js b/dist/index.js
index 12a4d02aac2b2364deb79d7785bb5b2d735d9e12..12a042dd383a3e91cbfe803af11210c5b40457a7 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -154,6 +154,17 @@ function processElement(element, headStyles, twi) {
       ...processedChildren
     );
   }
+  if (typeof modifiedElement.type === "function") {
+    const component = modifiedElement.type;
+    const renderedComponent = component(modifiedElement.props);
+    if (React.isValidElement(renderedComponent)) {
+      modifiedElement = processElement(
+        renderedComponent,
+        headStyles,
+        twi
+      );
+    }
+  }
   return modifiedElement;
 }
 function processHead(child, responsiveStyles) {
diff --git a/dist/index.mjs b/dist/index.mjs
index 2e58d8425e6bf5ed3928e99d20fd1e1006d70e36..3f7b69f98aaf9ab4c45c27153d55b774e6e326f8 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -121,6 +121,17 @@ function processElement(element, headStyles, twi) {
       ...processedChildren
     );
   }
+  if (typeof modifiedElement.type === "function") {
+    const component = modifiedElement.type;
+    const renderedComponent = component(modifiedElement.props);
+    if (React.isValidElement(renderedComponent)) {
+      modifiedElement = processElement(
+        renderedComponent,
+        headStyles,
+        twi
+      );
+    }
+  }
   return modifiedElement;
 }
 function processHead(child, responsiveStyles) {
```

**Comment by J4v4Scr1pt** [comment]
Still not working for me.
But my workaround with this is to wrap the component in Tailwind as well.. maybe not the best.. but it works for now.. 😅 

```
import { Img, Section, Tailwind } from '@react-email/components';
import * as React from 'react';
import tailwindConfig from '../../tailwind.config';

export const Header = () => (
	<Tailwind config={tailwindConfig}>
		<Section className="round flex h-[120px] items-center justify-center rounded-tl-lg rounded-tr-lg bg-wt-black">
			<Img
				src="src"
			/>
		</Section>
	</Tailwind>
);
```

**Comment by gabrielmfern** [comment]
> Still not working for me. But my workaround with this is to wrap the component in Tailwind as well.. maybe not the best.. but it works for now.. 😅
> 
> ```
> import { Img, Section, Tailwind } from '@react-email/components';
> import * as React from 'react';
> import tailwindConfig from '../../tailwind.config';
> 
> export const Header = () => (
> 	<Tailwind config={tailwindConfig}>
> 		<Section className="round flex h-[120px] items-center justify-center rounded-tl-lg rounded-tr-lg bg-wt-black">
> 			<Img
> 				src="src"
> 			/>
> 		</Section>
> 	</Tailwind>
> );
> ```

can you maybe give a reproduction so I can narrow down the problem on your case?

**Comment by J4v4Scr1pt** [comment]
In my case its just that when I import this in to my email-template the tailwind styles dont get applied to the component.

```
import {
	Body,
	Button,
	Column,
	Container,
	Head,
	Heading,
	Hr,
	Html,
	Img,
	Link,
	Preview,
	Row,
	Section,
	Tailwind,
	Text,
} from '@react-email/components';
import * as React from 'react';
import tailwindConfig from '../tailwind.config';
import { Header } from './components/Header';
import { Footer } from './components/Footer';

export const AssignedRole = () => {
	return (
		<Html>
			<Head />
			<Preview>{`Preview text`}</Preview>
			<Tailwind config={tailwindConfig}>
				<Body className="mx-auto my-auto rounded-lg bg-white font-sans">
					<Container className="mx-auto my-[40px] w-[600px] rounded-lg border border-solid border-[#eaeaea]"> // Styles work
						<Header /> // No styles without wrapping the component in Tailwind
					// ALLL THE CONTENT and all styles work
						<Footer /> // No styles without wrapping the component in Tailwind
					</Container>
				</Body>
			</Tailwind>
		</Html>
	);
};

export default AssignedRole;
```

**Comment by gabrielmfern** [comment]
> In my case its just that when I import this in to my email-template the tailwind styles dont get applied to the component.
> 
> ```
> import {
> 	Body,
> 	Button,
> 	Column,
> 	Container,
> 	Head,
> 	Heading,
> 	Hr,
> 	Html,
> 	Img,
> 	Link,
> 	Preview,
> 	Row,
> 	Section,
> 	Tailwind,
> 	Text,
> } from '@react-email/components';
> import * as React from 'react';
> import tailwindConfig from '../tailwind.config';
> import { Header } from './components/Header';
> import { Footer } from './components/Footer';
> 
> export const AssignedRole = () => {
> 	return (
> 		<Html>
> 			<Head />
> 			<Preview>{`Preview text`}</Preview>
> 			<Tailwind config={tailwindConfig}>
> 				<Body className="mx-auto my-auto rounded-lg bg-white font-sans">
> 					<Container className="mx-auto my-[40px] w-[600px] rounded-lg border border-solid border-[#eaeaea]"> // Styles work
> 						<Header /> // No styles without wrapping the component in Tailwind
> 					// ALLL THE CONTENT and all styles work
> 						<Footer /> // No styles without wrapping the component in Tailwind
> 					</Container>
> 				</Body>
> 			</Tailwind>
> 		</Html>
> 	);
> };
> 
> export default AssignedRole;
> ```

Could you try importing from `@react-email/tailwind` instead of components?

**Comment by J4v4Scr1pt** [comment]
I installed the Tailwind component and used it directly instead as you suggested but no styles get applied 🙁

**Comment by gabrielmfern** [comment]
> I installed the Tailwind component and used it directly instead as you suggested but no styles get applied 🙁

how did you install the patch?

**Comment by J4v4Scr1pt** [comment]
Ahh maybe that the issue. I just installed it like this: `pnpm add @react-email/tailwind -E`

`"prettier-plugin-tailwindcss": "^0.5.7",`

**Comment by gabrielmfern** [comment]
> Appreciate this temporary alternative. It's not perfect but a better workaround than wrapping multiple components around `<Tailwind />`

Hey @OussamaFadlaoui we have a new canary with a PR that should've fixed these kinds of issues, try out `@react-email/tailwind@0.0.13-canary.3` and any problems you have we will fix. Thanks for the patience!

**Comment by tomaszczura** [comment]
Unfortunately this still does not work for me event with `@react-email/tailwind` `v0.0.13` - styles are not applied to custom components

**Comment by lunoob** [comment]
> Unfortunately this still does not work for me event with `@react-email/tailwind` `v0.0.13` - styles are not applied to custom components

me too

**Comment by chris-trait** [comment]
Hey @gabrielmfern Is it not possible to use react hooks within Tailwind?

If I try using a React hook inside of my e-mail components wrapped in `<Tailwind>` then I get the rules of hooks error:

```
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
```

Minimal reproduction looks like:

```typescript
const TestA = () => {
    const [foo] = React.useState("hello")
    return <p>{foo}</p>
}
const html = render(
    <Tailwind>
        <TestA />
    </Tailwind>
)
```

The use case is to wrap a React context around the e-mail with common props and then consume them from some child components.

**Comment by gabrielmfern** [comment]
@chris-trait 
Hey, this is expected behavior, email templates will generally not work with hooks since they are supposed to be static,
all data that you want to add anywhere on it you should be passing through props to it before rendering with `@react-email/render`.
Hooks will work if you are rendering the email directly with React's rendering process but won't with our rendering process
because it generates static markup.

**Comment by chris-trait** [comment]
@gabrielmfern that does seem a little weird to me, as I do expect hooks to run for other "static" use cases like static site generation or server-side rendering, especially as async rendering is becoming more of a thing, and indeed hooks run fine with `renderToStaticMarkup`. React contexts also work fine in static rendering, the issue is with how the `tailwind` function traverses the component tree and calls components as plain functions: https://github.com/resendlabs/react-email/blob/2374d0d099360c63fe7427febb239776c3bd6b3b/packages/tailwind/src/tailwind.tsx#L95C32-L95C32.

For now my workaround is using `<MyReactContext.Consumer>{context => ...}</MyReactContext.Consumer>` instead of the hook.
